plugins/outputs/syslog/syslog_mapper.go

```diff
@@ -0,0 +1,193 @@
+package syslog
+
+import (
+	"errors"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/go-syslog/rfc5424"
+	"github.com/influxdata/telegraf"
+)
+
+type SyslogMapper struct {
+	DefaultSdid         string
+	DefaultSeverityCode uint8
+	DefaultFacilityCode uint8
+	DefaultAppname      string
+	Sdids               []string
+	Separator           string
+	reservedFields      map[string]bool
+}
+
+// MapMetricToSyslogMessage maps metrics tags/fields to syslog messages
+func (sm *SyslogMapper) MapMetricToSyslogMessage(metric telegraf.Metric) (*rfc5424.SyslogMessage, error) {
+	msg := &rfc5424.SyslogMessage{}
+
+	sm.mapPriority(metric, msg)
+	sm.mapStructuredData(metric, msg)
+	sm.mapAppname(metric, msg)
+	mapHostname(metric, msg)
+	mapTimestamp(metric, msg)
+	mapMsgID(metric, msg)
+	mapVersion(metric, msg)
+	mapProcID(metric, msg)
+	mapMsg(metric, msg)
```

Comment:
It seems that tags that aren't one of the known syslog tags are discarded?  Do you think we should add them as structured data?